src/field_theory/galois.lean

```diff
@@ -0,0 +1,230 @@
+  then `fixed_field (fixing_subgroup K) = K`
+Together, these two result prove the Galois correspondence
+-/
+
+noncomputable theory
+open_locale classical
+
+open finite_dimensional
+
+namespace galois
+
+section
+
+variables (F : Type*) [field F] (E : Type*) [field E] [algebra F E]
+
+/-- A field extension E/F is galois if it is both separable and normal -/
+@[class] def is_galois : Prop := is_separable F E ∧ normal F E
+
+instance is_galois_of_fixed_field (G : Type*) [group G] [fintype G] [mul_semiring_action G E] :
+  is_galois (mul_action.fixed_points G E) E :=
+⟨fixed_points.separable G E, fixed_points.normal G E⟩
+
+instance aut : group (E ≃ₐ[F] E) :=
+{ mul := λ ϕ ψ, ψ.trans ϕ,
+  mul_assoc := λ ϕ ψ χ, rfl,
+  one := 1,
+  one_mul := λ ϕ, by {ext, refl},
+  mul_one := λ ϕ, by {ext, refl},
+  inv := alg_equiv.symm,
+  mul_left_inv := λ ϕ, by {ext, exact alg_equiv.symm_apply_apply ϕ a} }
+
+lemma is_galois_implies_card_aut_eq_findim [finite_dimensional F E] [h : is_galois F E] :
+  fintype.card (E ≃ₐ[F] E) = findim F E :=
+begin
+  cases field.exists_primitive_element h.1 with α hα,
+  cases h.1 α with H1 h_separable,
+  cases h.2 α with H2 h_splits,
+  have switch : (⊤ : intermediate_field F E).to_subalgebra.to_submodule = ⊤ :=
+    by { ext, exact iff_of_true intermediate_field.mem_top submodule.mem_top },
+  rw [←findim_top, ←switch],
+  change fintype.card (E ≃ₐ[F] E) = findim F (⊤ : intermediate_field F E),
+  replace h_splits : polynomial.splits (algebra_map F F⟮α⟯) (minimal_polynomial H2),
+  { rw hα,
+    let map : E →+* (⊤ : intermediate_field F E) :=
+    { to_fun := λ x, ⟨x, intermediate_field.mem_top⟩,
+      map_one' := rfl,
+      map_mul' := λ _ _, rfl,
+      map_zero' := rfl,
+      map_add' := λ _ _, rfl },
+    rw (show algebra_map F (⊤ : intermediate_field F E) = map.comp (algebra_map F E),
+      by { ext, refl }),
+    exact polynomial.splits_comp_of_splits (algebra_map F E) map h_splits },
+  rw [←hα, @intermediate_field.findim_adjoin_integral F _ _ _ _ α H2,
+      ←@intermediate_field.alg_equiv_adjoin_integral F _ _ _ _ α H2 h_separable h_splits],
+  apply fintype.card_congr,
+  rw hα,
+  change (E ≃ₐ[F] E) ≃
+    ((⊤ : intermediate_field F E).to_subalgebra ≃ₐ[F] (⊤ : intermediate_field F E).to_subalgebra),
+  rw intermediate_field.top_to_subalgebra,
+  exact
+  { to_fun := λ ϕ, (algebra.top_equiv).trans (alg_equiv.trans ϕ (algebra.top_equiv).symm),
+    inv_fun := λ ϕ, (algebra.top_equiv).symm.trans (alg_equiv.trans ϕ (algebra.top_equiv)),
+    left_inv := λ _, by { ext, simp only [alg_equiv.apply_symm_apply, alg_equiv.trans_apply] },
+    right_inv := λ _, by { ext, simp only [alg_equiv.symm_apply_apply, alg_equiv.trans_apply] } },
+end
+
+end
+
+section galois_correspondence
+
+variables {F : Type*} [field F] {E : Type*} [field E] [algebra F E]
+variables (H : subgroup (E ≃ₐ[F] E)) (K : intermediate_field F E)
+
+instance is_galois_tower_top_of_is_galois [h : is_galois F E] : is_galois K E :=
```

Comment:
Same question as above: should this be a theorem rather than an instance?

Reply:
My sense is that if we're using [galois F E] rather than (h : galois F E), then these should be instances, but I'm not familar with mathlib conventions about this

Reply:
Yeah, my question is basically whether that's how we should be using it. It seems like making it an implicit rather than explicit hypothesis leads to annoying `haveI`s and such. But I'm really hoping that someone who knows more about the mathlib conventions will see this question and leave a comment.

Reply:
Ask on Zulip?

Reply:
https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Stating.20properties.20as.20an.20instance.20vs.2E.20a.20theorem

Reply:
Okay, looks like these should stay instances I guess.